pkg/egresspolicy/egresspolicy.go

```diff
@@ -0,0 +1,520 @@
+			}
+			delete(manager.policyPods, podData.id)
+			delete(manager.podDataStore, podData.id)
+		}
+	}
+
+	if upsertNew {
+		// Check if any of the current policies select this pod.
+		for _, config := range manager.policyConfigs {
+			if config.policyConfigSelectsPod(podData) {
+				manager.upsertPolicyPod(config, podData)
+			}
+		}
+	}
+}
+
+func (manager *Manager) OnUpdatePod(pod *slimcorev1.Pod, needsReassign bool, ready bool) {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+	// TODO add unit test to validate that we get callbacks only for relevant events
+	manager.OnUpdatePodLocked(pod, needsReassign || !ready, ready)
+}
+
+func (manager *Manager) OnDeletePod(pod *slimcorev1.Pod) {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+	if len(manager.policyConfigs) == 0 {
+		return
+	}
+	id := k8s.ServiceID{
+		Name:      pod.GetName(),
+		Namespace: pod.GetNamespace(),
+	}
+
+	podIPs := k8sUtils.ValidIPs(pod.Status)
+	if len(podIPs) == 0 {
+		return
+	}
+
+	var podData *podMetadata
+	var err error
+	if podData, err = manager.getPodMetadata(pod, podIPs); err != nil {
+		log.WithError(err).WithFields(logrus.Fields{
+			logfields.K8sPodName:   pod.Name,
+			logfields.K8sNamespace: pod.Namespace,
+		}).Error("failed to get valid pod metadata")
+		return
+	}
+
+	if policies, ok := manager.policyPods[id]; ok {
+		for _, policy := range policies {
+			config := manager.policyConfigs[policy]
+			manager.deleteEgressMap(config, podData)
+		}
+		delete(manager.policyPods, id)
+	}
+	delete(manager.podDataStore, id)
+}
+
+func (manager *Manager) getPodMetadata(pod *slimcorev1.Pod, podIPs []string) (*podMetadata, error) {
+	podId := k8s.ServiceID{
+		Name:      pod.GetName(),
+		Namespace: pod.GetNamespace(),
+	}
+
+	namespace, err := manager.getNamespaceForPod(pod)
+	if namespace == nil || err != nil {
+		log.Warn("Failed in getting namespace")
+		data := &podMetadata{
+			ips:    podIPs,
+			labels: pod.GetLabels(),
+			id:     podId,
+		}
+
+		// Keep id to pod data mapping up to date
+		manager.podDataStore[podId] = data
+		return data, nil
+	}
+
+	data := &podMetadata{
+		ips:      podIPs,
+		labels:   pod.GetLabels(),
+		nsLabels: namespace.GetLabels(),
+		id:       podId,
+	}
+
+	// Keep id to pod data mapping up to date
+	manager.podDataStore[podId] = data
+	return data, nil
+}
+
+func (m *Manager) getNamespaceForPod(pod *slimcorev1.Pod) (*slimcorev1.Namespace, error) {
+	nsName := pod.GetNamespace()
+	nsStore := m.storeGetter.GetStore("namespace")
+	nsItem, exists, err := nsStore.GetByKey(pod.GetNamespace())
+	if err != nil {
+		log.WithError(err).Warn("Failed to get namespace")
+		return nil, err
+	}
+	if !exists {
+		log.WithFields(logrus.Fields{"nsName": nsName}).Warn("Namespace does not exist")
+		return nil, nil
+	}
+	return nsItem.(*slimcorev1.Namespace), nil
+}
+
+// isValidConfig validates the given policy config.
+func (manager *Manager) isValidConfig(config EgressPolicyConfig) error {
+	return nil
+}
```

Comment:
Is this a TODO? Are you planning to implement it in this PR? If not, I would remove the calls to not give the impression anything is validated.

Reply:
I added the basic content for it to check that no duplicate egressIPs are configured. We will potentially have more logic to add to this function.

Thx for the comment.